Dockerfile

```diff
@@ -1,25 +1,45 @@
-FROM ruby:2.7.5-slim-buster
-RUN apt-get update -qq && apt-get upgrade -y && apt-get install -y build-essential nodejs && apt-get clean
+ARG base_image=ruby:2.7.5-slim-buster
+ 
+FROM $base_image AS builder
 
-# This image is only intended to be able to run this app in a production RAILS_ENV
-ENV RAILS_ENV production
+ENV RAILS_ENV=production GOVUK_APP_NAME=static
 
-ENV GOVUK_APP_NAME static
-ENV REDIS_URL redis://redis
-ENV PORT 3013
+RUN apt-get update -qq && \
+    apt-get upgrade -y && \
+    apt-get install -y build-essential nodejs && \
+    apt-get clean
 
-ENV APP_HOME /app
-RUN mkdir $APP_HOME
+RUN mkdir /app
 
-WORKDIR $APP_HOME
-ADD .ruby-version $APP_HOME/
-ADD Gemfile* $APP_HOME/
-RUN bundle config set deployment 'true'
-RUN bundle config set without 'development test'
-RUN bundle install --jobs 4
+WORKDIR /app
 
-ADD . $APP_HOME
+COPY Gemfile* .ruby-version /app/
+
+RUN bundle config set deployment 'true' && \
+    bundle config set without 'development test' && \
+    bundle install --jobs 4 --retry=2
+
+COPY . /app
 
 RUN GOVUK_WEBSITE_ROOT=https://www.gov.uk GOVUK_APP_DOMAIN=www.gov.uk bundle exec rails assets:precompile
 
+FROM $base_image
+ 
+ENV RAILS_ENV=production GOVUK_APP_NAME=static
+ 
+RUN apt-get update -qy && \
+    apt-get upgrade -y && \
+    apt-get install -y nodejs
+ 
+RUN groupadd -g 1001 appuser && \
+    useradd appuser -u 1001 -g 1001 --home /app
+ 
+COPY --from=builder /usr/local/bundle/ /usr/local/bundle/
+COPY --from=builder /app /app
+RUN mkdir -p /app/public/templates && chown -R 1001:1001 /app/public
```

Comment:
Does the app need to write to all of `/app/public` or just `/app/public/templates`?

Reply:
Good shout - it does not - updated with git amend